platform_channels/lib/src/counter_method_channel.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+/// This class includes implementation of two platform methods [increment],
+/// and [decrement] which are used to increment and decrement value
+/// of count respectively.
+class Counter {
+  /// Creates a [MethodChannel] with the specified name to invoke platform method.
+  /// In order to communicate across platforms, the name of MethodChannel
+  /// should be same on native and dart side.
+  static MethodChannel methodChannel = const MethodChannel('methodChannelDemo');
+
+  /// This method is responsible to increment and return the value of count.
+  static Future<int> increment({int counterValue}) async {
+    try {
+      // Invokes a platform method with given name and argument. The first
+      // argument to MethodChannel.invokeMethod represents the name of the
+      // platform method. The second argument represents the arguments passed
+      // to platform method.
+      final result = await methodChannel
+          .invokeMethod<int>('increment', {'count': counterValue});
+      return result;
+    } on Exception catch (_) {
+      return null;
```

Comment:
Why is the code trapping these exceptions? Would it not be better just to let them escape so that the UI can present the error message?

Reply:
Makes sense, using `catchError` now to show the specific error message.